### PR TITLE
Finish w_046 and import data

### PR DESCRIPTION
**Pull request by joshlory**
https://decomp.me/scratch/dKGxs
https://decomp.me/scratch/YGyQp

### CODE REVIEW

**Comment by bismurphy** [comment]
Any reason for creating the new Ext for weapon 46?

I was able to do a find and replace to make it all use the normal `ext.weapon` and it seems to match just as well:
https://decomp.me/scratch/nR04G

**Comment by joshlory** [comment]
> Any reason for creating the new Ext for weapon 46?

Ah I modified `ET_Weapon` in the context before copying it to a new `ET_WeaponUnk046` struct. I wasn't able to get it matching with the existing struct.

File: src/weapon/w_046.c [hunk 1]
```diff
@@ -9,9 +13,368 @@ extern s32 g_HandId;
...
+            prim->r2 = a1 * 0xC0 / 256;
+            prim->g2 = a1 * 0x60 / 256;
+            prim->b2 = a1 * 0x60 / 256;
+
+            prim->r1 = prim->g1 = prim->b1 = 0;
+            prim->r3 = prim->g3 = prim->b3 = 0;
+
+            a1 = var_i >> 1;
+            a0 = a1 + 1;
+            var_a0 = D_146000_8017ABD0[a0 - ((a0 / 16) * 0x10)];
```
[the remark this answers is not in the document]
Comment (joshlory) [reply src/weapon/w_046.c:339]: Helpful, thank you! Worth opening an issue on https://github.com/matt-kempster/m2c/issues to identify more of these automatically?

**Comment by bismurphy** [comment]
> > Any reason for creating the new Ext for weapon 46?
> 
> Ah I modified `ET_Weapon` in the context before copying it to a new `ET_WeaponUnk046` struct. I wasn't able to get it matching with the existing struct.

Got it, silly me, should have checked on that.

Sad that it doesn't match up, these weapons are super weird. Looks good then, at least for now while we're still learning weapons. Once we get all the weapons decompiled hopefully we can find a unified solution to the Ext problem for them.

File: src/weapon/w_046.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (bismurphy) [reply src/weapon/w_046.c:339]: I don't really know anything about m2c but sure, if you would want to open an issue I don't see why that would be a problem

[the remark this answers is not in the document]
Comment (joshlory) [reply src/weapon/w_046.c:339]: Done! https://github.com/matt-kempster/m2c/issues/282

File: src/weapon/w_046.c [hunk 2]
```diff
@@ -1,6 +1,10 @@
 // Weapon ID #46. Used by weapons:
 // Holbein dagger, Blue knuckles
 #include "weapon_private.h"
+extern u16 D_146000_8017A2B0[];
+extern u16 D_146000_8017A530[];
+extern u16 D_146000_8017A670[];
+extern u16 D_146000_8017A3F0[];
 extern u16* g_WeaponCluts[];
 extern s32 g_HandId;
 #include "shared.h"
```
Comment (Xeeynamo) [inline src/weapon/w_046.c:10]:
```suggestion
```
you should be able to delete these three lines

**Review by Xeeynamo — COMMENTED** [review]
Looking great! Since you imported the data, could you link this weapon? There is an example here: https://github.com/Xeeynamo/sotn-decomp/pull/1493/files

File: src/weapon/w_046.c (hunk 2 above)
Comment (joshlory) [reply src/weapon/w_046.c:10]: I think they're referenced in `shared.h` — should I move the includes down below the data defines?

...but I can get rid of these lines!

```c
extern u16 D_146000_8017A2B0[];
extern u16 D_146000_8017A530[];
extern u16 D_146000_8017A670[];
extern u16 D_146000_8017A3F0[];
```

Comment (Xeeynamo) [reply src/weapon/w_046.c:10]: Yeah, you would need to move `shared.h` right before `static void EntityWeaponAttack(Entity* self)` like the PR I linked above

Not sure about the `extern u16 D_146000_*`, you should be able to delete them!

**Comment by joshlory** [comment]
In order to get the linking working I had to decompile the last `EntityWeaponAttack` func in the file. Should be good to go hopefully!

**Pull request merged**